src/Hl7.Fhir.Core/Rest/FhirClient.cs

```diff
@@ -1133,10 +1133,19 @@ private void verifyServerVersion()
                 throw Error.NotSupported("Cannot read the conformance statement of the server to verify FHIR version compatibility");
             }
 
-            if (!ModelInfo.CheckMinorVersionCompatibility(conf.Version))
+            try
+            {
+                if (!ModelInfo.CheckMinorVersionCompatibility(conf.Version))
+                {
+                    throw Error.NotSupported($"This client supports FHIR version {ModelInfo.Version} but the server uses version {conf.Version}");
+                }
+            }
+            catch (ArgumentNullException)
```

Comment:
Is een `if(conf.Version == null)` hierboven niet effectiever dan eerst een null aan `CheckMinorVersionCompatibility` voeren, om daarna te hoopen dat er een ArgumentNullException komt?  Mocht iemand later binnen `CheckMinorVersionCompatibility` bedenken dat die exception anders moet, dan loopt je het gevaar dat dit faalt, terwijl een directe if altijd blijft werken....